src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let arrayOfMen = people.filter(person => person.sex === 'm');
+
+  arrayOfMen = century
+    ? arrayOfMen.filter(men => Math.ceil(men.died / 100) === century)
+    : arrayOfMen;
+
+  const ageOfMan = arrayOfMen.map(person => person.died - person.born);
+
+  const sumOfAge = ageOfMan.reduce((sum, age) => sum + age, 0);
+
+  const averageAgeOfMen = sumOfAge / arrayOfMen.length;
```

Comment:
create separate function that takes array of people, and return average age, and use it here, and in the next function